spec/datadog/profiling/native_extension_spec.rb

```diff
@@ -197,4 +197,64 @@ def wait_for_thread_to_die
       end
     end
   end
+
+  describe 'is_current_thread_holding_the_gvl' do
+    subject(:is_current_thread_holding_the_gvl) do
+      Datadog::Profiling::NativeExtension::Testing._native_is_current_thread_holding_the_gvl
+    end
+
+    context 'when current thread is holding the global VM lock' do
+      it { is_expected.to be true }
+    end
+
+    context 'when current thread is not holding the global VM lock' do
+      subject(:is_current_thread_holding_the_gvl) do
+        Datadog::Profiling::NativeExtension::Testing._native_release_gvl_and_call_is_current_thread_holding_the_gvl
+      end
+
+      it { is_expected.to be false }
+    end
+
+    describe 'correctness' do
+      before do
+        skip 'Ruby 2.2 does not expose ruby_thread_has_gvl_p so nothing to compare to' if RUBY_VERSION.start_with?('2.2.')
```

Comment:
What of Ruby 2.1?

Reply:
The profiler no longer supports 2.1 -- we spit out a warning instead ( https://github.com/DataDog/dd-trace-rb/pull/2140 ) and don't even try to compile the native extension.